ch2.3.1-algorithm-t/README.md

```diff
@@ -0,0 +1,333 @@
+Algorithm T has something slightly arbitrary to it.
+All these jumps; and that stack you have to keep around.
+And yet, the recursive version of the inorder traversal is so clean, so beautiful:
+
+```typescript
+function traverse(btree, visit) {
+    if (btree !== Λ) {
+        traverse(btree.llink, visit);
+        visit(btree);
+        traverse(btree.rlink, visit);
+    }
+}
+```
+
+About the only bad thing I can say about this implementation is that it could overflow the stack.
+Not going to happen with a balanced binary tree, but could well happen with a pathological one, such as a couple thousand rlinks in a row.
+
+Is there a way we could translate the above recursive version into the iterative version in Algorithm T?
+Preferably, you know, step by step.
```

Comment:
Remove 'you know'?

Reply:
"If you argue with your editor, you've already lost." And yet, dammit, I read it again, and I like it better with that "you know" still in there.

I could go through all the reasons I do. Deliberate informality. Breaking "the fourth wall" a little. Bringing length and rhythm to a sentence which would otherwise be too short. _Winking_ at the reader, suggesting that doing the translation in a single step would be valid but much less satisfactory whereas a step-wise solution would help elucidate the process, and author and reader are both agreeing on that.

A "musing" gets to be informal enough to call the reader "you", so I don't see the problem there either.

But, ach! I will change it. If it snagged you up during review, then that means there's a better way to phrase it. Kill thy darlings. :+1: